pkg/kinflate/commands/diff_test.go

```diff
@@ -0,0 +1,119 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package commands
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/ghodss/yaml"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubectl/pkg/kinflate/util/fs"
+)
+
+type DiffTestCase struct {
+	Description string   `yaml:"description"`
+	Args        []string `yaml:"args"`
+	Filename    string   `yaml:"filename"`
+	// path to the file that contains the expected output
+	ExpectedDiff string `yaml:"expectedDiff"`
+}
+
+func TestDiff(t *testing.T) {
+	const updateEnvVar = "UPDATE_KINFLATE_EXPECTED_DATA"
+	updateKinflateExpected := os.Getenv(updateEnvVar) == "true"
+
+	noopDir, _ := regexp.Compile(`/tmp/noop-[0-9]*/`)
+	transformedDir, _ := regexp.Compile(`/tmp/transformed-[0-9]*/`)
+	timestamp, _ := regexp.Compile(`[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1]) (2[0-3]|[01][0-9]):[0-5][0-9]:[0-5][0-9].[0-9]* [+-]{1}[0-9]{4}`)
+
+	fs := fs.MakeRealFS()
+
+	var (
+		name     string
+		testcase DiffTestCase
+	)
+
+	testcases := sets.NewString()
+	filepath.Walk("testdata", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if path == "testdata" {
+			return nil
+		}
+		name := filepath.Base(path)
+		if info.IsDir() {
+			if strings.HasPrefix(name, "testcase-") {
+				testcases.Insert(strings.TrimPrefix(name, "testcase-"))
+			}
+			return filepath.SkipDir
+		}
+		return nil
+	})
+	// sanity check that we found the right folder
+	if !testcases.Has("simple") {
+		t.Fatalf("Error locating kinflate inflate testcases")
+	}
+
+	for _, testcaseName := range testcases.List() {
+		t.Run(testcaseName, func(t *testing.T) {
+			name = testcaseName
+			testcase = DiffTestCase{}
+			testcaseDir := filepath.Join("testdata", "testcase-"+name)
+			testcaseData, err := ioutil.ReadFile(filepath.Join(testcaseDir, "test.yaml"))
+			if err != nil {
+				t.Fatalf("%s: %v", name, err)
+			}
+			if err := yaml.Unmarshal(testcaseData, &testcase); err != nil {
+				t.Fatalf("%s: %v", name, err)
+			}
+
+			diffOps := &diffOptions{
+				manifestPath: testcase.Filename,
+			}
+			buf := bytes.NewBuffer([]byte{})
+			err = diffOps.RunDiff(buf, os.Stderr, fs)
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+			actualString := string(buf.Bytes())
+			actualString = noopDir.ReplaceAllString(actualString, "/tmp/noop/")
+			actualString = transformedDir.ReplaceAllString(actualString, "/tmp/transformed/")
+			actualString = timestamp.ReplaceAllString(actualString, "YYYY-MM-DD HH:MM:SS")
+			actualBytes := []byte(actualString)
+			if !updateKinflateExpected {
```

Comment:
All of this is kind of sad. Can we replace the diff binary with "true" for the test, and just check that the files are what we expect?

Reply:
The files are deleted once `RunDiff` exits. So in testing, we don't have those files available.